### PR TITLE
[INJIMOB-2369] authorization endpoint discovery through auth server wellknown

### PR DESCRIPTION
**Pull request by KiruthikaJeyashankar**
## Description

> Authorization endpoint is identified through oauth-authorization-server metadata rather hardcoding the authorization_endpoint to be `${authorization_server[0]}/authorize`

More in detail,
- authorization server property's first entry of well-known is used to construct the url of oauth-authorization-server wellknown
- This is used to get the server metadata, post this wallet compatibility check is performed. As of now, check involves grant_types_supported to have authorization_code
- If check passes, authorization happy flow is followed else grant type not supported error screen is displayed
- Additionally, in case of any error like n/w error, server error or any technical error during api request to oauth-authorization-server wellknown,  error screen is displayed accordingly

## Issue ticket number and link

> [INJIMOB-2369](https://mosip.atlassian.net/browse/INJIMOB-2369?focusedCommentId=95255)

## Screenshots

> Successful Vc download through authorization endpoint discovert

image

[INJIMOB-2369]: https://mosip.atlassian.net/browse/INJIMOB-2369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: machines/Issuers/IssuersMachine.ts [hunk 1]
```diff
@@ -496,4 +535,5 @@ export interface issuerType {
   credential_configurations_supported: object;
   display: [displayType];
   credentialTypes: [CredentialTypes];
+  authorizationEndpoint: string;
```
Comment (swatigoel) [inline machines/Issuers/IssuersMachine.ts:538]: authorization_servers cam be removed as not in use now

File: shared/api.ts [hunk 2]
```diff
@@ -170,6 +184,21 @@ export const CACHED_API = {
       fetchCall: API.fetchIssuerWellknownConfig.bind(null, issuerId),
     }),
 
+  fetchIssuerAuthorizationServerMetadata: (
+    authorizationServerUrl: string,
+    isCachePreferred: boolean = false,
+  ) =>
+    generateCacheAPIFunction({
+      isCachePreferred,
+      cacheKey: API_CACHED_STORAGE_KEYS.fetchIssuerWellknownConfig(
```
Comment (swatigoel) [inline shared/api.ts:193]: this cache key is same as of fetch issuer wellknown.

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: machines/Issuers/IssuersMachine.ts (hunk 1 above)
Comment (Alka1703) [reply machines/Issuers/IssuersMachine.ts:538]: Done

File: shared/api.ts (hunk 2 above)
Comment (KiruthikaJeyashankar) [reply shared/api.ts]: Modified cache key to be specific to issuer authorization server metadata

**Review by swatigoel — APPROVED** [review]

**Pull request merged**